packages/dialog/src/DialogBase.ts

```diff
@@ -150,9 +148,13 @@ export class DialogBase extends FocusVisiblePolyfillMixin(SpectrumElement) {
 
     protected override update(changes: PropertyValues<this>): void {
         if (changes.has('open') && changes.get('open') !== undefined) {
-            this.transitionPromise = new Promise(
-                (res) => (this.resolveTransitionPromise = res)
-            );
+            this.animating = true;
```

Comment:
This sort of async bookkeeping & synchronization is what suggests that perhaps we should consider a [state machine](https://www.youtube.com/watch?v=RqTxtOXcv8Y) pattern. (/cc @najikahalsema)

Reply:
An this state management might not "be cleaned up" but a new overlay approach, as it's bound deeply into the content that is being overlaid, which means we might still benefit from it...

Reply:
For purposes of extensibility it might be nice to consider moving away from this lifecycle approach where the consumer is having to manage transitions or consolidating the logic into an OverlayBase element so that if a client wants to plugin to the Overlay API with their own custom component they can do so rather easily without have to rewrite a lot of this animation state management logic. #2086